### PR TITLE
[ADD] analytics_piwik

### PR DESCRIPTION
**Pull request by hbrunn**
# Piwik analytics

This module allows to simply configure you odoo website for using Piwik_.

.. _Piwik: http://piwik.org
# Configuration

To configure this module, you need to:
- go to Settings/Configuration/Website Settings
- fill in `Piwik website ID` and `Piwik host`
- Take care that you fill in the correct value for `Piwik host`, it should be
  so that http[s]://[`Piwik host`]/piwik.php resolves to your copy of `piwik.php`

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Why not website_analytics_piwik?

**Comment by hbrunn** [comment]
I can do that if you want

**Comment by pedrobaeza** [comment]
I think it's a better name that implies the "area".

**Comment by hbrunn** [comment]
done

File: website_analytics_piwik/models/website_config_settings.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#
+##############################################################################
+from openerp import models, fields, api
+
+
+class WebsiteConfigSettings(models.Model):
+    _inherit = 'website.config.settings'
+
+    piwik_analytics_id = fields.Integer(
+        related = ['website_id', 'piwik_analytics_id'])
```
Comment (pedrobaeza) [inline website_analytics_piwik/models/website_config_settings.py]: Curious form of putting a related field. I always put it this way: `related="website_id.piwik_analytics_id"`

Anyways, don't put spaces in a keyword argument.

File: website_analytics_piwik/views/templates.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+<openerp>
+    <data>
+        <template id="layout" inherit_id="website.layout">
+            <xpath expr="html/body" position="inside">
+                <script t-if="website and website.piwik_analytics_id and website.piwik_analytics_host" type="text/javascript">
+                    var _paq = _paq || [];
+                    _paq.push(['trackPageView']);
+                    _paq.push(['enableLinkTracking']);
+                    (function() {
+                    var u="//<t t-esc="website.piwik_analytics_host" />/";
```
Comment (pedrobaeza) [inline website_analytics_piwik/views/templates.xml]: t-esc or t-raw?

File: website_analytics_piwik/models/website_config_settings.py (hunk 1 above)
Comment (hbrunn) [inline website_analytics_piwik/models/website_config_settings.py]: it's the new way to do it (tm): https://github.com/odoo/odoo/blob/8.0/openerp/fields.py#L451

File: website_analytics_piwik/views/templates.xml (hunk 2 above)
Comment (hbrunn) [inline website_analytics_piwik/views/templates.xml]: yes, raw makes more sense here, thanks!

File: website_analytics_piwik/models/website_config_settings.py (hunk 1 above)
Comment (pedrobaeza) [inline website_analytics_piwik/models/website_config_settings.py]: I still feels more readable the other way, but I'm saying only as opinion, not to make you to change it.

**Comment by pedrobaeza** [comment]
:+1: 

Thanks

**Comment by lmignon** [comment]
:+1:  Thanks

**Comment by antespi** [comment]
:+1:

**Pull request merged**